packages/core/src/editor.tsx

```diff
@@ -36,6 +53,54 @@ export function Editor<K extends string = string>(props: EditorProps<K>) {
   }
 }
 
+export const EditorProvider: React.FunctionComponent<{
+  omitDragDropContext?: boolean
+}> = props => {
+  const store = React.useMemo(() => {
+    return createStore({
+      instances: {}
+    }).store
+    // We want to create the store only once
+    // TODO: add effects that handle changes to plugins and defaultPlugin (by dispatching an action)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  const children = <Provider store={store}>{props.children}</Provider>
+  if (props.omitDragDropContext) return children
+  return (
+    <DragDropContextProvider backend={HTML5Backend}>
+      {children}
+    </DragDropContextProvider>
+  )
+}
+
+export function EditorInstance<K extends string = string>(
+  props: EditorProps<K> & InnerEditorProps
+) {
+  const { store } = React.useContext(EditorContext)
+  const isMainInstance = React.useRef(
```

Comment:
Never updated, so probably might not work as expected. Not sure what this is intended to do @Knorrke? I _suppose_ that you want the `React.useEffect` to depend on `props.mirror` instead and calculate `isMainInstance` inside the effect (might also depend on `store.getState()[props.scope]`?)

Reply:
This was a hack... the problem is, that checking if the state was already initialized in `React.useEffect` (sometimes?) wouldn't work , as the state is initialized in another useEffect. So I wanted to check this during render time, but need it only once...

Reply:
Fixed in https://github.com/edtr-io/edtr-io/pull/136/commits/96fa20b65e5adb13e78d82da7d7b236373757aed